marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs

```diff
@@ -155,8 +155,8 @@ class Flow {
     }
     let stepDetails = this.step.getStepByNameAndType(stepRef.name, stepRef.type);
 
-    this.globalContext.targetDb = stepRef.targetDatabase || stepDetails.targetDatabase || this.globalContext.targetDb;
-    this.globalContext.sourceDb = stepRef.sourceDatabase || stepDetails.targetDatabase || this.globalContext.sourceDb;
+    this.globalContext.targetDb = stepRef.targetDatabase || (stepDetails !== undefined &&  stepDetails.targetDatabase) || this.globalContext.targetDb;
+    this.globalContext.sourceDb = stepRef.sourceDatabase || (stepDetails !== undefined && stepDetails.targetDatabase) || this.globalContext.sourceDb;
```

Comment:
I am not sure if this check is required

Reply:
oh, i forgot removing the conditions when i tested...